File: src/main/java/de/mm/fileshuffler/filter/MusicFilter.java
package de.mm.fileshuffler.filter;

import java.io.File;
import java.io.FileFilter;

public class MusicFilter implements FileFilter {

	@Override
	public boolean accept(File path) {
		
		if (path.isDirectory()) {
			return true;
		}
		
		String[] parts = path.getPath().split("\\.");
		String ending = parts[parts.length - 1];
		
		switch (ending) {
			case "mp3":
				return true;
			default:
				return false;
		}
	}
	
	@Override
	public String toString() {
		return "MusicFilter";
	}

}

File: src/main/kotlin/de/mm/fileshuffler/filter/MusicFilter.kt
package de.mm.fileshuffler.filter

import java.io.File
import java.io.FileFilter

class MusicFilter : FileFilter {

	override fun accept(path: File): Boolean {

		if (path.isDirectory) {
			return true
		}

		val parts = path.path.split("\\.".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
		val ending = parts[parts.size - 1]

		when (ending) {
			"mp3" -> return true
			else -> return false
		}
	}

	override fun toString(): String {
		return "MusicFilter"
	}

}
